### PR TITLE
build: add component update action

### LINKED ISSUE #784: create re-usable action for updating main chart
Ideally we could reuse the code to update the main chart across all of our projects.

### PR DESCRIPTION
**Pull request by rokroskar**
Adds an actions directory with generic re-usable actions. 

Two actions are added: 

* action to update a component version in the main renku chart
* action to build images and publish the chart with chartpress

closes #784

### CODE REVIEW

File: actions/update-component-version/update-upstream.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# update the upstream repo
+cd upstream-repo
+
+# set up git
+git config --global user.email "$GIT_EMAIL"
+git config --global user.name "$GIT_USER"
+
+# update the chart requirements and push
+git checkout -b auto-update/${CHART_NAME}-${CHART_VERSION} master
+yq w -i charts/renku/requirements.yaml "dependencies.(name==${CHART_NAME}).version" $CHART_VERSION
```
Comment (jachro) [inline actions/update-component-version/update-upstream.sh]: I'm wondering if requiring UPSTREAM_REPO variable is something we should do assuming that we have to hardcode paths like charts/renku/requirements.yaml anyway?

Comment (rokroskar) [reply actions/update-component-version/update-upstream.sh]: oh yeah good point - I could get rid of that

Comment (rokroskar) [reply actions/update-component-version/update-upstream.sh]: or, alternatively, make the path to the requirements file configurable

Comment (jachro) [reply actions/update-component-version/update-upstream.sh]: Yes, you can try and then we'll see how complex that would become. Do you aim to end up with something others can use? To me, that's something very renku specific but of course I could be wrong.

Comment (rokroskar) [reply actions/update-component-version/update-upstream.sh]: yes, this one is probably pretty renku-specific - the other one (chartpress) is pretty generic

Comment (rokroskar) [reply actions/update-component-version/update-upstream.sh]: ok I simplified it a bit

**Comment by rokroskar** [comment]
I think there's still a problem unfortunately - looks like 

```
if [[ $GITHUB_REF =~ "tags" ]]
```

is not valid shell syntax (needs bash)...

**Comment by jachro** [comment]
Yes, I have `#!/bin/env bash` in my travis scripts.

**Comment by rokroskar** [comment]
Right, I just need to install bash in the docker image.

**Comment by jachro** [comment]
Does it work better with bash?

**Comment by rokroskar** [comment]
yeah no more errors: https://github.com/SwissDataScienceCenter/renku-python/runs/460751439?check_suite_focus=true#step:6:10

**Pull request merged**